apps/lib/apps.c

```diff
@@ -299,6 +299,11 @@ static char *app_get_pass(const char *arg, int keepbio)
              * Can't do BIO_gets on an fd BIO so add a buffering BIO
              */
             btmp = BIO_new(BIO_f_buffer());
+            if (btmp == NULL) {
+                BIO_free(pwdbio);
```

Comment:
It's unclear if this should be deleted or not, but it might depend on keepbio. If it gets deleted, it should probably use BIO_free_all() instead of BIO_free(), and it should set pwdbio to NULL since it's a static variable.

Reply:
Note that there is similar code below that does not delete on error. Both should probably do the same.

Reply:
The block is under `if (!keepbio || pwdbio == NULL) {` which is at apps.c:273, so I delete it.

Reply:
And in fact, the condition `if (!keepbio || pwdbio == NULL)` at apps.c:273 seems meaningless? As pwdbio` is declared at `apps:258` by `static BIO *pwdbio = NULL;`, so `pwdbio` is NULL.